src/js/components/DataTable/Searcher.js

```diff
@@ -59,6 +60,7 @@ class Searcher extends Component {
           </Box>
         ) : null}
         <Button
+          data-testid={`searcher-on-filtering-${property}`}
```

Comment:
better name?

Reply:
Why do we need to add these into the production code?

Can we instead make the test smarter and get the component by traversing the DOM a bit?

Reply:
https://github.com/coderas/babel-plugin-jsx-remove-data-test-id
We can remove `data-testid` for production thanks to this babel plugin (if use).

I think it is fragile to test a complicated component using the DOM structure.
It's hard to understand the real DOM structure calculated by React.

Of course I can test it without `data-testid`. How do you think @ericsoderberghp ?

Reply:
friendly reminder to @ericsoderberghp :)

Reply:
My hesitation to add this is two-fold: 1) I'm reticent to add something to the production DOM. We try to keep the DOM as clean as possible. 2) We aren't using this pattern anywhere else. My suggestion instead of this is to add an `a11yTitle` to the Button like "Search ${property}" and then have the test search for the component with that `aria-label`.

Reply:
I replaced `data-testid` with `a11yTitle` (for Button) and `name` (for TextInput).